client/sites/components/sites-dataviews/dataviews-fields/site-field.tsx

```diff
@@ -60,7 +63,49 @@ const ListTileTitle = styled.div`
 	align-items: center;
 `;
 
-const SiteField = ( { site, openSitePreviewPane }: Props ) => {
+/**
+ * Renders an anchor element that can trigger analytics calls
+ * or client-side router navigation via the `onNavigate` prop.
+ */
+const Link = ( {
+	disabled,
+	href,
+	onNavigate,
+	...props
+}: {
+	/** Accessibly disable the link. Use sparingly. */
+	disabled?: boolean;
+	/** Called when the user wants to navigate to the link. */
+	onNavigate?: ( shouldOpenNewTab: boolean, event: React.MouseEvent ) => void;
```

Comment:
Not sure if I love this function signature for `onNavigate`. Might need to change after we look at more use cases.

I think the name is ok though? We want a name that makes it clear that this is more abstract than an `onClick`.

Reply:
Sounds good to me, at least for now, exactly because it separates from "click" events.

Also, it aligns with the new history-related APIs that some browsers already support (https://developer.chrome.com/docs/web-platform/navigation-api/)